Dynatrace/dynatrace-operator
# Add clean up logic for version and connection-info reconcilers

## Description

In previous refactors we added a few TODOs about cleaning up after ourselves. 

Here, I added/improved the cleanup for 2 reconcilers + 1:
- version
- connection-info
- +1, kubesystemUUID now is only set once per DynaKube as it should never change, this way we save a request per reconcile

The improvement is not final as:
- ActiveGate conditions are missing, as they are added in https://github.com/Dynatrace/dynatrace-operator/pull/2923
   - So the AG connection info clean-up is not as "clean" as the OA connection-info (only ping the API if there is actually something to clean up)
- Istio reconciler still needs to be updated
- authToken reconciler still needs to be updated (it also will have its own condition, and might make sense to fix the cleanup in that PR)

I think this is a big enough change for now, further improvements will have separate PRs. 
- (easier to review, and if we want a different "style" of doing this, I would prefer to not have to rework everything 😅 ) 

## How can this be tested?

Deploy dynakubes, and turn on/off OA and AG. See that the related conditions and secrets are removed/created.
- only version, and connection-info related stuff should be affected

## Checklist
